tfx/benchmarks/benchmark_base.py

```diff
@@ -19,22 +19,82 @@
 
 from absl import flags
 import apache_beam as beam
+from apache_beam.pipeline import PipelineOptions
 from tensorflow.python.platform import test  # pylint: disable=g-direct-tensorflow-import
+from tfx.benchmarks import mode_config
 
 FLAGS = flags.FLAGS
 flags.DEFINE_string(
     "beam_runner", "DirectRunner",
     "Beam runner to use - any runner name accepted by "
     "apache_beam.runners.create_runner")
 
-
 class BenchmarkBase(test.Benchmark):
+  """Base class for running Beam pipelines on various runners"""
 
-  def _create_beam_pipeline(self):
+  def __init__(self):
+    super(BenchmarkBase, self).__init__()
+    self.beam_pipeline_mode = mode_config.DEFAULT_MODE
+    self.num_workers = 1
+    self.cloud_dataflow_temp_loc = None
+
+  def _set_cloud_dataflow_flags(self):
```

Comment:
Could you use [`PipelineOptions`](https://beam.apache.org/releases/pydoc/2.22.0/apache_beam.options.pipeline_options.html#apache_beam.options.pipeline_options.PipelineOptions) instead of flags? The flags get parsed into `PipelineOptions` by Beam anyway, so this gets rid of the need to do string construction and interpolation (which is more error prone).

Reply:
Done, I switched everything to use PipelineOptions instead of flags.